pkg/api/v1beta1/types.go

```diff
@@ -609,6 +615,8 @@ type Endpoints struct {
 	// "UDP".  Defaults to "TCP".
 	Protocol  Protocol `json:"protocol,omitempty" description:"IP protocol for endpoint ports; must be UDP or TCP; TCP if unspecified"`
 	Endpoints []string `json:"endpoints" description:"list of endpoints corresponding to a service, of the form address:port, such as 10.10.1.1:1909"`
+	// Optional: The kubernetes object related to the entry point.
+	TargetRefs []EndpointObjectReference `json:"targetRefs,omitempty" description:"list of references to objects providing the endpoints"`
```

Comment:
I didn't want to assume that all the Endpoints had to specify an ObjectReference. In the case one wouldn't exist we would have to pad with some nil value in order to preserve the positional relationship.